apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notices.js

```diff
@@ -0,0 +1,185 @@
+/* global wpcomGlobalStyles */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { ExternalLink, Notice } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createInterpolateElement, render, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useCanvas } from './use-canvas';
+import { useGlobalStylesConfig } from './use-global-styles-config';
+import './notice.scss';
+
+const trackEvent = ( eventName, isSiteEditor = true ) =>
+	recordTracksEvent( eventName, {
+		context: isSiteEditor ? 'site-editor' : 'post-editor',
+		blog_id: wpcomGlobalStyles.wpcomBlogId,
+	} );
+
+function GlobalStylesWarningNotice() {
+	const { globalStylesInUse } = useGlobalStylesConfig();
+
+	useEffect( () => {
+		if ( globalStylesInUse ) {
+			trackEvent( 'calypso_global_styles_gating_notice_view_canvas_show' );
+		}
+	}, [ globalStylesInUse ] );
+
+	if ( ! globalStylesInUse ) {
+		return null;
+	}
+
+	return (
+		<Notice status="warning" isDismissible={ false } className="wpcom-global-styles-notice">
+			{ createInterpolateElement(
+				__(
+					'Your site includes customized styles that are only visible to visitors after <a>upgrading to the Premium plan or higher</a>.',
+					'full-site-editing'
+				),
+				{
+					a: (
+						<ExternalLink
+							href={ wpcomGlobalStyles.upgradeUrl }
+							target="_blank"
+							onClick={ () =>
+								trackEvent( 'calypso_global_styles_gating_notice_view_canvas_upgrade_click' )
+							}
+						/>
+					),
+				}
+			) }
+		</Notice>
+	);
+}
+
+function GlobalStylesViewNotice() {
+	const { canvas } = useCanvas();
+
+	useEffect( () => {
+		if ( canvas !== 'view' ) {
+			return;
+		}
+
+		const footer = document.querySelector( '.edit-site-sidebar__footer' );
+		if ( ! footer ) {
+			return;
+		}
+
+		const noticeContainer = document.createElement( 'div' );
+		footer.prepend( noticeContainer );
+		render( <GlobalStylesWarningNotice />, noticeContainer );
+	}, [ canvas ] );
+
+	return null;
+}
+
+function GlobalStylesEditNotice() {
+	const NOTICE_ID = 'wpcom-global-styles/gating-notice';
+	const { globalStylesInUse, globalStylesId } = useGlobalStylesConfig();
+	const { isSiteEditor, isPostEditor } = useSelect(
+		( select ) => ( {
+			isSiteEditor: !! select( 'core/edit-site' ),
+			isPostEditor: ! select( 'core/edit-site' ) && !! select( 'core/editor' ).getCurrentPostId(),
+		} ),
+		[]
+	);
+
+	const { createWarningNotice, removeNotice } = useDispatch( 'core/notices' );
+	const { editEntityRecord } = useDispatch( 'core' );
+
+	const upgradePlan = useCallback( () => {
+		window.open( wpcomGlobalStyles.upgradeUrl, '_blank' ).focus();
+		trackEvent( 'calypso_global_styles_gating_notice_upgrade_click', isSiteEditor );
+	}, [ isSiteEditor ] );
+
+	const previewSite = useCallback( () => {
+		window.open( wpcomGlobalStyles.previewUrl, '_blank' ).focus();
```

Comment:
When in Post Editor mode, I think the preview URL should be the post itself, not the main page. Otherwise, the user doesn't have a way to see how their page/post would look like without styles.

Reply:
I agree that's a better approach, but it's not that trivial to do so because we first need to save the post and then preview it. We basically need to port/duplicate this logic: https://github.com/WordPress/gutenberg/blob/83a33e7c24ee1db8fe426052ac73137f51289234/packages/editor/src/components/post-preview-button/index.js#L149

While doable, I'm reluctant about making this PR more complex, so can we maybe keep the site preview for now and add the post preview in a immediate follow-up? Alternatively, I can also remove the preview button.

Reply:
I think it makes sense to remove the Preview button for now and make a follow-up PR for it. :)